### ISSUE TITLE
API: dashbaords db overwrite: saving dashboard failed: UNIQUE constraint failed: dashboard.org_id, dashboard.folder_id, dashboard.title

### ISSUE THREAD

**Issue by baryluk** [grafana/grafana#69301]
### 
# What went wrong?

I had existing dashboard from previous versions. They were generated using Python, then uploaded via API to grafana.

Recently our Grafana was updated to newer version, and script stopped working.

I am getting error 500, with body: `{"message":"Failed to save dashboard","traceID":""}`

Looking at grafana logs I see:

```
logger=context userId=9 orgId=1 uname=sa-autogen-1-wb-dashboard-generator t=2023-05-31T09:59:19.536952278Z
  level=error msg="Failed to save dashboard"
  error="saving dashboard failed: UNIQUE constraint failed: dashboard.org_id, dashboard.folder_id, dashboard.title"
  remote_addr=10.1.1.1 traceID=
logger=context userId=9 orgId=1 uname=sa-autogen-1-wb-dashboard-generator t=2023-05-31T09:59:19.536996697Z
  level=error msg="Request Completed"
  method=POST path=/api/dashboards/db status=500
  remote_addr=10.1.1.1 time_ms=29 duration=29.083281ms size=51 referer= handler=/api/dashboards/db
```
****

### 
# How do we reproduce it?

```
$ cat >request.json <<-EOF
{"overwrite":true,"dashboard":
{
  "__inputs": [],
  "annotations": {
    "list": []
  },
  "description": "Autogenerated",
  "editable": true,
  "gnetId": null,
  "graphTooltip": 0,
  "hideControls": false,
  "id": null,
  "links": [],
  "panels": [
  ],
  "refresh": "5m",
  "rows": [],
  "schemaVersion": 12,
  "sharedCrosshair": true,
  "style": "dark",
  "tags": [
    "ixc"
  ],
  "templating": {
    "list": [
    ]
  },
  "time": {
    "from": "now-3h",
    "to": "now"
  },
  "timepicker": {
    "hidden": false,
    "refresh_intervals": [
      "5s",
      "10s",
      "30s",
      "1m",
      "5m",
      "15m",
      "30m",
      "1h",
      "2h",
      "1d"
    ],
    "time_options": [
      "5m",
      "15m",
      "1h",
      "6h",
      "12h",
      "24h",
      "2d",
      "7d",
      "30d"
    ]
  },
  "timezone": "utc",
  "title": "my_dashboard3",
  "uid": null,
  "version": 0
}
}
EOF
$
```

```
$ curl \
              --request POST \
              --header "Accept: application/json" \
              --header "Content-Type: application/json" \
              --header "Authorization: Bearer ${MY_GRAFANA_API_KEY}" \
              --data "@request.json" \
              --write-out '\nHTTP status: %{http_code}\n' \
              "https://grafana.my.example.com/api/dashboards/db"
```

Note, this only fails, if the dashboard existed before in pre-v9.5.2

Step to reproduce would be to run v9.4.x, run above snippet. Upgrade to v9.5.2, then run above again.

### 
# What Grafana version are you using?

v9.5.2 (cfcea75916)

### 
### Grafana Platform?

Kubernetes, OpenShift on bare metal

### Is this a Regression?

Yes. It was working in 9.4.x

### Workaround

Deleting dashboard via UI, and reuploading solves the problem. Only issue with that is that URL is now new and existing links and bookmarks are broken.

**Comment by usmangt** [comment]
Hi @baryluk 

I think the possible reason could be that the Dashboard `schemaVersion` is not the same (incompatible) but let me forward this one to the Engineers as they might confirm more about what is causing this.

**Comment by Joey777210** [comment]
Encountered similar issue. After upgrade from 7.x to 9.4, and migrate graph(old) to new timeseries, db can not be saved

**Comment by sakjur** [comment]
This seems to be the underlying issue patched by https://github.com/grafana/grafana/pull/70723 .
I've written a bit about what went wrong in https://github.com/grafana/grafana/issues/66373#issuecomment-1610970520 . In short, the underscore in your title is what's causing this problem, because a function was using the slug in place of the title to find an existing dashboard. We've patched that for the next release of Grafana, which I believe will be v10.0.2 later this month, and it's being backported to the next v9.5.x release as well (I don't know when that'll be released, unfortunately).

I'll close this as fixed by #70723, poke me if you think that is not correct 🙂
